### PR TITLE
Fix branch

### PR DESCRIPTION
**Pull request by MikitaDubovik**
Fixed issues that were been found after review.

### CODE REVIEW

File: Module.Introduction/Controllers/CategoriesApiController.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        private readonly ICategoriesService _categoriesService;
+
+        public CategoriesApiController(ICategoriesService categoriesService)
+        {
+            _categoriesService = categoriesService;
+        }
+
+        // GET: api/Api
+        [Route("")]
```
Comment (Lexani) [inline Module.Introduction/Controllers/CategoriesApiController.cs]: you can remove this route attribute cause it is empty

File: Module.Introduction/Controllers/CategoriesApiController.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    public class CategoriesApiController : ControllerBase
+    {
+        private readonly ICategoriesService _categoriesService;
+
+        public CategoriesApiController(ICategoriesService categoriesService)
+        {
+            _categoriesService = categoriesService;
+        }
+
+        // GET: api/Api
```
Comment (Lexani) [inline Module.Introduction/Controllers/CategoriesApiController.cs]: strange comment, is this really the route to api?

File: Module.Introduction/Controllers/CategoriesApiController.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        public async Task<IActionResult> GetImage(int id)
+        {
+            var ms = await _categoriesService.GetImage(id);
+            return File(ms.ToArray(), "image/jpeg");
+        }
+
+        [Route("image")]
+        [HttpPost]
+        public async Task<IActionResult> Edit(int id, IFormFile file)
+        {
```
Comment (Lexani) [inline Module.Introduction/Controllers/CategoriesApiController.cs:79]: I suggest moving retrieving all logic for retrieving and updating picture to the service

File: Module.Introduction/Controllers/CategoriesApiController.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                return BadRequest();
+            }
+        }
+
+        private bool CategoriesExists(int id)
+        {
+            return _categoriesService.GetAllAsync().Result.Any(e => e.CategoryId == id);
```
Comment (Lexani) [inline Module.Introduction/Controllers/CategoriesApiController.cs]: do you have method getById? Do you need to retrieve all categories from db?

File: Module.Introduction/Controllers/ProductsController.cs [hunk 5]
```diff
@@ -141,7 +151,7 @@ public async Task<IActionResult> Delete(int? id)
                 return NotFound();
             }
 
-            var products = await _context.Products
+            var products = await _productsService.GetAllAsync().Result.AsQueryable()
```
Comment (Lexani) [inline Module.Introduction/Controllers/ProductsController.cs]: Here you are trying to get only one product (FirstOrDefaultAsync) with id but fetching all , why?

File: Module.Introduction/Services/FilesService.cs [hunk 6]
```diff
@@ -10,35 +10,43 @@ namespace Module.Introduction.Services
...
+                if (maximumNumberOfImage < 1)
+                {
+                    maximumNumberOfImage = 2;
+                }
+
+                if (numberOfFiles < maximumNumberOfImage)
                 {
                     await File.WriteAllBytesAsync(
-                        $@"D:\Work\CachedImages\CachedImage{categories.CategoryId}.txt",
+                        _nameOfDirectory + $@"\CachedImage{categories.CategoryId}.txt",
```
Comment (Lexani) [inline Module.Introduction/Services/FilesService.cs:42]: is it working without .txt extension?)

Comment (MikitaDubovik) [reply Module.Introduction/Services/FilesService.cs:42]: > is it working without .txt extension?)

No, it isn't

**Pull request merged**